mollie/api/objects/list.py

```diff
@@ -84,16 +69,87 @@ def has_previous(self):
         """Return True if the ObjectList contains an url for the previous set."""
         return self._get_link("previous") is not None
 
+    @abstractmethod
+    def get_next(self):
+        ...
+
+    @abstractmethod
+    def get_previous(self):
+        ...
+
+    @property
+    @abstractmethod
+    def object_type(self):
+        ...
+
+    @abstractmethod
+    def new(self, result):
+        ...
+
+
+
+class PaginationList(ListBase):
+    _parent: "ResourceBase"
+
+    def __init__(self, result, parent: "ResourceBase", client=None):
+        # If an empty dataset was injected, we mock the structure that the remainder of the clas expects.
+        # TODO: it would be better if the ObjectList was initiated with a list of results, rather than with
+        # the full datastructure as it is now, so we can remove all this mucking around with fake data,
+        # mocked result objects, and loads of lengthy accessor workarounds everywhere in the ObjectList.
+        self._parent = parent
+
+        if result == {}:
+            result = {"_embedded": {f"{self._parent.object_type.get_object_name()}": []}, "count": 0}
```

Comment:
Because we have a reference to the parent, we can now still infer the correct name of the collection, but it should probably be verified that `get_object_name` indeed matches the expected name in the responses.

Reply:
The method `get_object_name` is defined to return the correct name for the object. When auto-generation fails, it's overriden to provide the correct value, for instance at: https://github.com/mollie/mollie-api-python/blob/66afd4e6f827b7345eab11a19da1c855c5835ad9/mollie/api/objects/payment_link.py#L11-L13
Any unexpected return value is a bug. You can trust the method to return the correct value.